### ISSUE TITLE
ResourcePoolsBuilder.heap usability issue

### ISSUE THREAD

**Issue by mathieucarbou** [ehcache/ehcache3#1296]
ResourcePoolsBuilder contains both heap method:
- One that is static (1 argument), shortcut for the builder
- One that is not static and takes 2 arguments

This causes some really bad behavior and usability issues because we CAN actually call a static method from a class instance. So all these examples will compile fine:

1.

```
      CacheConfiguration<String, String> config1 = CacheConfigurationBuilder.newCacheConfigurationBuilder(
          String.class, String.class,
              heap(10)
              .offheap(1, MemoryUnit.MB)
              .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
          .build()
```

2.

```
      CacheConfiguration<String, String> config1 = CacheConfigurationBuilder.newCacheConfigurationBuilder(
          String.class, String.class,
              newResourcePoolsBuilder()
                  .heap(10, EntryUnit.ENTRIES)
              .offheap(1, MemoryUnit.MB)
              .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
          .build()
```

3.

```
      CacheConfiguration<String, String> config1 = CacheConfigurationBuilder.newCacheConfigurationBuilder(
          String.class, String.class,
              newResourcePoolsBuilder()
              .offheap(1, MemoryUnit.MB)
              .heap(10)
              .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
          .build()
```

4.

```
      CacheConfiguration<String, String> config1 = CacheConfigurationBuilder.newCacheConfigurationBuilder(
          String.class, String.class,
              newResourcePoolsBuilder()
              .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB))
              .heap(10))
          .build()
```

5.

```
      CacheConfiguration<String, String> config1 = CacheConfigurationBuilder.newCacheConfigurationBuilder(
          String.class, String.class,
              heap(10)
              .offheap(1, MemoryUnit.MB)
              .heap(15, ENTRIES)
              .heap(5)
              .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB)))
          .build()
```

etc...
- 1, 2 are working.
- 3, 4, 5, etc with any call to `.heap(int)` during the building phase is not working.

The issue is that the static `.heap(int)` method creates a new builder and discards all the previously set configuration.

CC @anthonydahanne @GaryWKeim

**Comment by chrisdennis** [comment]
This really feels like a specialized version of the "builder never used" problem.  I propose we file and 'fix' that problem using a logging finalizer.

**Comment by ljacomet** [comment]
Will be addressed through #1483

**Comment by chrisdennis** [comment]
See @ljacomet comment above. I don't think this can be solved without breaking backwards compatibility. Fix to #1483 would solve the issue.
